.bash_nav

```diff
@@ -7,3 +7,33 @@ alias desk='cd /mnt/c/Users/gsera/Desktop'
 # source bash
 
 alias srcbash='source ~/.bashrc'
+
+# ---This alias effectively creates a bookmarking app---
+# it adds global symlink shortcuts for use with the 'cds' command
+# ex: 'cds studycloud' will run 'cd -P ~/.short/studycloud'
+# to add a shortcut: go to the target directory and run 'cds + <shortcut_name>' or 'cds + .' to use the current directory name
+# to remove a shortcut: run 'cds - <shortcut_name>' (you can also tack on multiple shortcut names separated by spaces)
+# to list your shortcuts: run 'cds .'
+cds() {
+    if [ ! -d ~/.short ]; then
+        mkdir ~/.short
+    fi
+    if [ $# -eq 1 ] && [ $1 = "." ]; then
+        ls ~/.short/
+    else
+        if [ $# -eq 2 ] && [ $1 = "+" ]; then
+            ln -s "$PWD" "$HOME/.short/$2"
+        elif [ $# -ge 2 ] && [ $1 = "-" ]; then
+            local args=( ${@:2} )
+            eval "rm ${args[@]/#/"~/.short/"}"
+        else
+            if [ $# -eq 0 ] || [ -d ./$1 ] || [ $1 = "-" ]; then
+                cd "$@"
+            elif [ -d "$(realpath ~/.short/$1)" ]; then
```

Comment:
Otherwise, we'll want to actually jump through a shortcut. First, let's check that there exists a shortcut with the specified name. We can use the realpath command to give us the full, absolute path to a directory.
This is helpful when the user uses directory chaining in their first argument. Thus, `cds hw/exercises` will also check that the `exercises` directory exists inside of the target of the `hw` shortcut.